### PR TITLE
Extend Hotspot Model with Status Field

### PR DESCRIPTION
**Pull request by AnhTuDo1998**
PR for extending Hotspot model with status field, as discuss in #51

### CODE REVIEW

File: src/models/hotspot_status.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct Status {
```
Comment (madninja) [inline src/models/hotspot_status.rs:4]: put the status definition in the hotspot model file

File: src/models/hotspot_status.rs [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    /// The online status of the hotspot
+    pub online: Option<String>,
+    /// The listening IP address of the hotspot
+    pub listen_addrs: Option<Vec<String>>,
+    /// The height of hotspot
+    pub height: Option<u64>,
```
Comment (madninja) [inline src/models/hotspot_status.rs:10]: Add the timestamp the status. Height and other information is based on when the hotspot sent out its gossip data so you need to represent height at the right time

File: src/models/hotspot_status.rs (hunk 1 above)
Comment (AnhTuDo1998) [reply src/models/hotspot_status.rs]: Noted with thanks!

File: src/models/hotspot_status.rs (hunk 2 above)
Comment (AnhTuDo1998) [reply src/models/hotspot_status.rs]: I have added timestamp as part of Status struct.

File: src/models/hotspot.rs [hunk 3]
```diff
@@ -42,6 +43,20 @@ pub struct Hotspot {
...
     /// The current reward scale for the hotspot
     pub reward_scale: Option<f64>,
+    /// The current status for the hotspot
+    pub status: Status,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct Status {
+    /// The timestamp of hotspot status, when it last sends gossip
+    pub timestamp: DateTime<Utc>,
```
Comment (madninja) [inline src/models/hotspot.rs:53]: Timestamp is an `Option<DateTime<Utc>>`

**Review by madninja — CHANGES_REQUESTED** [review]

File: src/models/hotspot.rs (hunk 3 above)
Comment (AnhTuDo1998) [reply src/models/hotspot.rs:53]: Noted. Just curious, when would the API return null datetime such that we would need Option in this case?

Comment (madninja) [reply src/models/hotspot.rs]: when the hotspot is offline is when we can see a status field with null values. For example

https://api.helim.io/v1/hotspots/13DTPjZeaJ8YKhrNyNuFdY8aCXHEpFuEUy7Qd2UDzLEQQqBi6sk

returns status as:

```
        "status": {
            "height": null,
            "listen_addrs": null,
            "online": "offline",
            "timestamp": null
        }
```

Comment (AnhTuDo1998) [reply src/models/hotspot.rs]: I see, thank's for the clarification. I have made the change as needed.

**Review by madninja — APPROVED** [review]

**Comment by madninja** [comment]
Thank you!

**Pull request merged**